src/settings/toml/deploy_target.rs

```diff
@@ -0,0 +1,152 @@
+use crate::settings::toml::Route;
+
+impl DeployTarget {
+    pub fn build(
+        script_name: &str,
+        route_config: &RouteConfig,
+    ) -> Result<DeployTarget, failure::Error> {
+        if route_config.workers_dev_false_by_itself() || route_config.has_conflicting_targets() {
```

Comment:
i think i'd argue that `workers_dev_false_by_itself` should be lumped in with `has_conflicting_targets`. Could keep it as its own function and just call it from the `has_conflicting_targets` function. Or maybe a wrapper function that does this `||` called `route_config.is_valid()`

Reply:
workers_dev_false_by_itself is not a conflicting target situation tho; conflicting target implies more than one target is specified, rather than zero targets are specified. i'm inclined to leave it for now.

Reply:
sure - would `is_valid` be ok? i just like when there are single functions so you dont have to remember to call both of them and do an or. not gonna fight for it too hard but still think it's worthwhile